src/lib/zAuction.ts

```diff
@@ -87,6 +87,39 @@ export async function getBidsForNft(
 	return bids;
 }
 
+export async function getBidsListForNfts(
+	baseApiUri: string,
+	contract: string[],
+	tokenId: string[],
+) {
+	const nftIdsArray: string[] = [];
+	const bidsList: string[] = [];
+
+	for (let i = 0; i < tokenId.length; i++) {
+		// it builds the array of nftIds strings
+		nftIdsArray.push(getNftId(contract[i], tokenId[i]));
+	}
+
+	const nftIdsBody = { nftIds: nftIdsArray }; //builds the correct body to send
+
+	let endpoints = getApiEndpoints(baseApiUri);
+
+	const response = await fetch(endpoints.bidListEndpoint, {
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' },
+		body: JSON.stringify(nftIdsBody),
+	});
+
+	const jsonResponse = await response.json(); //awaits the json of the response
+
+	for (let i = 0; i < tokenId.length; i++) {
+		//builds an array of only bid lists
+		bidsList.push(jsonResponse[i].bids);
```

Comment:
By the logic of your function `bidsList` is a `string[]` and you are pushing `jsonResponse[i].bids` into the array of string...

Therefore you'd think that `jsonResponse[i].bids` is a string, but I'm almost certain it is not.